ext/date/tests/getSetMicroseconds.phpt

```diff
@@ -0,0 +1,197 @@
+--TEST--
+Tests for DateTime[Immutable]::[get|set]Microseconds
+--FILE--
+<?php
+
+class MyDateTime extends DateTime {};
+class MyDateTimeImmutable extends DateTimeImmutable {};
+
+$microsecondsList = array(
+    0,
+    999999,
+    -1,
+    1000000
+);
+
+$dt = new DateTime('2023-04-05T06:07:08.901234+00:00');
+$dti = new DateTimeImmutable('2023-04-05T06:07:08.901234+00:00');
+$myDt = new MyDateTime('2023-04-05T06:07:08.901234+00:00');
+$myDti = new MyDateTimeImmutable('2023-04-05T06:07:08.901234+00:00');
+
+echo 'DateTime::getMicroseconds(): ' . var_export($dt->getMicroseconds(), true) . "\n";
+var_dump($dt);
+
+echo 'DateTimeImmutable::getMicroseconds(): ' . var_export($dti->getMicroseconds(), true) . "\n";
+var_dump($dti);
+
+echo 'MyDateTime::getMicroseconds(): ' . var_export($myDt->getMicroseconds(), true) . "\n";
+var_dump($myDt);
+
+echo 'MyDateTimeImmutable::getMicroseconds(): ' . var_export($myDti->getMicroseconds(), true) . "\n";
+var_dump($myDti);
+
+foreach ($microsecondsList as $microseconds) {
+    echo "##################################\n";
+
+    echo 'DateTime::setMicroseconds('.var_export($microseconds, true).'):';
+    try {
+        var_dump($dt->setMicroseconds($microseconds));
+    } catch (Throwable $e) {
+        echo get_class($e) . ' ' . $e->getMessage() . "\n";
+    }
+    echo 'DateTime::getMicroseconds(): ' . var_export($dt->getMicroseconds(), true) . "\n";
+
+    echo 'DateTimeImmutable::setMicroseconds('.var_export($microseconds, true).'):';
+    try {
+        var_dump($dti->setMicroseconds($microseconds));
+    } catch (Throwable $e) {
+        echo get_class($e) . ' ' . $e->getMessage() . "\n";
+    }
+    echo 'DateTimeImmutable::getMicroseconds(): ' . var_export($dti->getMicroseconds(), true) . "\n";
+
+    echo 'MyDateTime::setMicroseconds('.var_export($microseconds, true).'):';
+    try {
+        var_dump($myDt->setMicroseconds($microseconds));
+    } catch (Throwable $e) {
+        echo get_class($e) . ' ' . $e->getMessage() . "\n";
+    }
+    echo 'MyDateTime::getMicroseconds(): ' . var_export($myDt->getMicroseconds(), true) . "\n";
+
+    echo 'MyDateTimeImmutable::setMicroseconds('.var_export($microseconds, true).'):';
+    try {
+        var_dump($myDti->setMicroseconds($microseconds));
+    } catch (Throwable $e) {
+        echo get_class($e) . ' ' . $e->getMessage() . "\n";
+    }
+    echo 'MyDateTimeImmutable::getMicroseconds(): ' . var_export($myDti->getMicroseconds(), true) . "\n";
+}
+
```

Comment:
Please close all phpt files with `?>` so that you can run them with `sapi/cli/php ext/datetests/getSetMicroseconds.phpt`.